src/main/java/mods/railcraft/common/blocks/tracks/outfitted/kits/TrackKitSpeedTransition.java

```diff
@@ -67,6 +72,18 @@ public void onMinecartPass(EntityMinecart cart) {
                         slowOrNormalCartSpeed(cart, highSpeed);
                     }
                 }
+            } else if (dir == BlockRailBase.EnumRailDirection.EAST_WEST) {
+                if (world.isSideSolid(getPos().west(), EnumFacing.EAST)) {
+                    cart.motionX = START_BOOST;
+                } else if (world.isSideSolid(getPos().east(), EnumFacing.WEST)) {
+                    cart.motionX = -START_BOOST;
+                }
+            } else if (dir == BlockRailBase.EnumRailDirection.NORTH_SOUTH) {
+                if (world.isSideSolid(getPos().north(), EnumFacing.SOUTH)) {
+                    cart.motionZ = START_BOOST;
+                } else if (world.isSideSolid(getPos().south(), EnumFacing.NORTH)) {
+                    cart.motionZ = -START_BOOST;
+                }
```

Comment:
This could probably share some code with the Booster if you are going to add it.